docs/badges/distribution.rst

```diff
@@ -0,0 +1,311 @@
+
+Both options perform the same:
+
+1. Pull available for Organization badge templates from `Credly API`_:
+
+.. code-block::
+
+    GET /organizations/<organization_id>/badge_templates
+
+2. Create ``CredlyBadgeTemplate`` records for each available template.
+
+Badge templates synchronization
+###############################
+
+After the initial Credly badge templates creation on Open edX Credentials side, system starts listen to their updates via available web hooks:
+
+- ``badge_template.created`` (event is triggered whenever a badge template is created)
+    - system creates new ``CredlyBadgeTemplate`` in inactive state;
+    - badges are not start being processed until manual activation;
+- ``badge_template.changed`` (event is triggered whenever a badge template is changed)
+    - system updates corresponding ``CredlyBadgeTemplate`` relevant attributes;
+    - badges processing continues;
+- ``badge_template.deleted`` (event is triggered whenever a badge template is delete)
+    - system archives corresponding ``CredlyBadgeTemplate``;
+    - badges processing stops;
+
+Requirements setup
+~~~~~~~~~~~~~~~~~~
+
+Created badges `must be configured`_.
+
+Desired Requirements are created for each badge template manually:
+
+- create a Requirement for a single course with specific ID for completion event.
+- create a Requirement for a single course with specific ID for revocation event (reverse effect).
+
+CCX courses
+###########
+
+    Both event types (for passing course grade and failing course grade) carry additional information for custom courses identification.
+
+Awarded badges
+~~~~~~~~~~~~~~
+
+Already awarded badges are stored as `UserCredential`_ records.
+
+Credly backend manages its own extended version of user credential:
+
+.. code-block::
+
+    CredlyBadge(UserCredential):
+        """
+        Earned badge.
+        """
+        - - - unified attrs - - -
+        ...
+        - uuid
+        - username: <username>
+        - status: "awarded" | "revoked"
+        ...
+
+        - - - specific attrs - - -
+
+        - issued_badge_id: <uuid>
+        - issued_at: <timestamp>
+        - accept_badge_url: <URL>
+
+
+Side effects
+~~~~~~~~~~~~
+
+`Badge Collector`_ emits signals when:
+
+- a badge is `awarded`_
+- a badge is `revoked`_
+
+There are events which may be used as triggers for distribution **side effects**:
+
+- on award:
+    - credentials.apps.badges.distribution.credly.handlers.notify_user_awarding
+    - credentials.apps.badges.distribution.credly.handlers.issue_badge
+- on revoke:
+    - credentials.apps.badges.distribution.credly.handlers.notify_user_revocation
+    - credentials.apps.badges.distribution.credly.handlers.revoke_badge
+
+Badge issuance
+##############
+
+Badge issuance side effect is performed via `issue a badge Credly API`_ endpoint.
+Required fields:
+
+.. code-block::
+
+    recipient_email: <user-email>
+    issued_to_first_name: <str>
+    issued_to_last_name: <str>
```

Comment:
Names are hard, should we be looking at the `full_name` fields versus First/Last? Or is this a requirement on the Credly side?

Reply:
[Yeap](https://www.credly.com/docs/issued_badges)